### PR TITLE
Rename swap params to improve devUX

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Description

This always drove me a little nuts. And when thinking about issue #894, I realized it was difficult to understand the current state by just looking at the code.

There are (for good reasons) two separate swap parameter structs that we pass around. (The passing around of memory structs is not ideal, but necessary due to stack-too-deep limitations.)

One corresponds to the "external" API from the router: with token addresses and amounts in native token decimals. The other is the "internal" API used within the Vault (corresponding to `IBasePool`); e.g., to enforce consistency when calling the main operation and various hooks. This uses scaled values, indices instead of addresses, etc.

I've always found this a bit hard to read. It's confusing to have `swapParams` and `poolSwapParams`, especially when it's sometimes called `params`, and we also have `SwapState` with completely generic variables like `state`. It's especially confusing when we're converting between types.

This PR adds a block comment explaining why we have two "SwapParam" things, and what each does. It renames SwapParams to VaultSwapParams, so that we now have VaultSwapParams for the "Vault API" and PoolSwapParams for the "Pool/Hook API." It also explicitly names all the generic `params` and `state` variables so that we can keep them all straight.

As for the choice of names, I suppose VaultSwapParams could be RouterSwapParams; I just thought that might be confusing in the Vault. We could also say ExternalSwapParams and InternalSwapParams, but these aren't very specific. PoolSwapParams was already there, and it does mean "api for IBasePool and pool hooks," so I left it alone.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [X] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [X] The diff is legible and has no extraneous changes
- [N/A] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [X] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

**Review by jubeira — COMMENTED** [review]
I think this makes sense for swap given that we have pool / vault swap params coexisting very closely together in the code. The change itself looks good.

@danielmkm do you think this improves readability a bit?

File: pkg/interfaces/contracts/vault/VaultTypes.sol [hunk 1]
```diff
@@ -196,17 +196,26 @@ enum SwapKind {
...
 }
 
+// There are two "SwapParams" structs defined below. `VaultSwapParams` corresponds to the external swap API defined
+// in the Router contracts, which uses explicit token addresses, the amount given and limit on the calculated amount
+// expressed in native token decimals, and optional user data passed in from the caller.
+//
+// `PoolSwapParams` passes some of this information through (kind, userData), but "translates" the parameters to fit
+// the internal swap API used by `IBasePool`. It scales amounts to full 18-decimal precision, adds the token balances,
+// converts the raw token addresses to indices, and adds the address of the router originating the request. It does
+// not need the limit, since this is checked at the Router level.
```
Comment (joaobrunoah) [inline pkg/interfaces/contracts/vault/VaultTypes.sol:206]: Nice!

**Review by joaobrunoah — APPROVED** [review]
Makes sense!

**Review by jubeira — APPROVED** [review]
Let's go!

**Pull request merged**